psd-web/app/services/delete_user.rb

```diff
@@ -0,0 +1,25 @@
+class DeleteUser
+  include Interactor
+
+  def call
+    context.fail!(error: "No user supplied") unless context.user
+    context.fail!(error: "User already deleted") if context.user.deleted?
+
+    ActiveRecord::Base.transaction do
+      context.user.mark_as_deleted!
+      assign_user_investigations_to_their_team
+    end
+  end
+
+private
+
+  def assign_user_investigations_to_their_team
+    context.team = context.user.teams.first
+
+    context.user.investigations.each do |investigation|
+      investigation.assignable = context.team
+      investigation.save
```

Comment:
I think the job ought to be raising an exception if the save fails?

```suggestion
      investigation.save!
```

Reply:
This was my implicit suggestion ;)

I think we need to have a conversation about this `record.save` or `record.update` it's something I see very often in pull requests and can potentially have disastrous side effects/hard to track bugs down the line.

Reply:
You guys are right! Sloppy Marc!